### DISCUSSION TITLE
Certificate Request Error

### DISCUSSION THREAD

**Discussion by marvk** [community/community#23410] [Other Feature Feedback, Questions, & Ideas]
<p>GitHub is unable to create a certificate for my custom domain GitHub Pages page. This has worked before, but recently I noticed an certificate error on the page. After reading the documentation, it seemed like I was missing a <code>www CNAME marvk.github.io</code> DNS entry.</p>
<p>After adding this entry, the configuration error on the projects GitHub Pages configuration site got resolved, but I am left with a seemingly properly configured domain, but no certificates being generated.</p>
<p>These are my DNS entries:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/9/1/910cfd15eb4bd6a4fed056f2f76804e7227b9ba9.png" data-download-href="/uploads/short-url/kHb3NeHBnTKVlHJTvmLNWsN1vaV.png?dl=1" title="" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/6569856/181076792-e9fade8f-148d-4831-9afb-460b2f832a67.png" alt="" data-base62-sha1="kHb3NeHBnTKVlHJTvmLNWsN1vaV" width="690" height="414" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/9/1/910cfd15eb4bd6a4fed056f2f76804e7227b9ba9_2_690x414.png, https://aws1.discourse-cdn.com/github/optimized/3X/9/1/910cfd15eb4bd6a4fed056f2f76804e7227b9ba9_2_1035x621.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/9/1/910cfd15eb4bd6a4fed056f2f76804e7227b9ba9.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/9/1/910cfd15eb4bd6a4fed056f2f76804e7227b9ba9_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename"></span><span class="informations">1273×764 25.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>And at the content root I have a CNAME file with the content <code>vatprism.org</code></p>
<p>Barring DNS propagation, which seems to have already happened, I fail to see the issue. Should I just try giving it a bit more time or is there some issue I’m missing?</p>
<p>Thanks in advance.</p>

**Comment by yoannchaudet** [comment]
<p>Hello and welcome here!</p>
<p>You could not have asked at a better time. We actually deployed changes today around certificates handling to correct an entire class of errors that were happening lately.</p>
<p>Your DNS configuration (on the screenshot) looks good. You probably removed it because I am unable to query these records anymore (that would prevent us from obtaining a certificate for your domain). If you want to give it another try, I will be happy to monitor this certificate provisioning for you.</p>

**Comment by marvk** [comment]
<p>Hey there, thank you! Other than TTL I haven’t touched the records in about 36 hours. The A records have been untouched for months, the only thing I added about 36 hours ago was the CNAME.</p>
<p>On <a href="http://whatismydns.com" rel="noopener nofollow ugc">whatismydns.com</a> the propagation seems poor:</p>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://www.whatsmydns.net/?utm_source=whatsmydns.com&amp;utm_medium=redirect#A/vatprism.org">
  <header class="source">
      <img src="https://www.whatsmydns.net/favicon.ico" class="site-icon" width="48" height="48">

      <a href="https://www.whatsmydns.net/?utm_source=whatsmydns.com&amp;utm_medium=redirect#A/vatprism.org" target="_blank" rel="noopener nofollow ugc">whatsmydns.net</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/345;"><img src="https://user-images.githubusercontent.com/6569856/181076793-38282622-e6f3-4d44-8b10-89552661e107.png" class="thumbnail" width="690" height="345" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/5/59de7c2b434ea92fc358d2b6d4328e3decd69b66_2_690x345.png, https://aws1.discourse-cdn.com/github/optimized/2X/5/59de7c2b434ea92fc358d2b6d4328e3decd69b66_2_1035x517.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/5/59de7c2b434ea92fc358d2b6d4328e3decd69b66_2_1380x690.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/5/59de7c2b434ea92fc358d2b6d4328e3decd69b66_2_10x10.png"></div>

<h3><a href="https://www.whatsmydns.net/?utm_source=whatsmydns.com&amp;utm_medium=redirect#A/vatprism.org" target="_blank" rel="noopener nofollow ugc">DNS Checker - DNS Propagation Check &amp; DNS Lookup</a></h3>

  <p>Instant DNS Propagation Check. Global DNS Lookup and DNS Propagation Checker. Check your DNS records around the world.</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://www.whatsmydns.net/?utm_source=whatsmydns.com&amp;utm_medium=redirect#CNAME/www.vatprism.org">
  <header class="source">
      <img src="https://www.whatsmydns.net/favicon.ico" class="site-icon" width="48" height="48">

      <a href="https://www.whatsmydns.net/?utm_source=whatsmydns.com&amp;utm_medium=redirect#CNAME/www.vatprism.org" target="_blank" rel="noopener nofollow ugc">whatsmydns.net</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/345;"><img src="https://user-images.githubusercontent.com/6569856/181076794-7c96cf8d-f3ac-470f-8577-d33321e15452.png" class="thumbnail" width="690" height="345" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/5/59de7c2b434ea92fc358d2b6d4328e3decd69b66_2_690x345.png, https://aws1.discourse-cdn.com/github/optimized/2X/5/59de7c2b434ea92fc358d2b6d4328e3decd69b66_2_1035x517.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/5/59de7c2b434ea92fc358d2b6d4328e3decd69b66_2_1380x690.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/5/59de7c2b434ea92fc358d2b6d4328e3decd69b66_2_10x10.png"></div>

<h3><a href="https://www.whatsmydns.net/?utm_source=whatsmydns.com&amp;utm_medium=redirect#CNAME/www.vatprism.org" target="_blank" rel="noopener nofollow ugc">DNS Checker - DNS Propagation Check &amp; DNS Lookup</a></h3>

  <p>Instant DNS Propagation Check. Global DNS Lookup and DNS Propagation Checker. Check your DNS records around the world.</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>However, on <a href="http://dnschecker.com" rel="noopener nofollow ugc">dnschecker.com</a> propagation is close to exhaustive:</p>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://dnschecker.org/#A/vatprism.org">
  <header class="source">
      <img src="https://user-images.githubusercontent.com/6569856/181076795-9be1b998-9adf-4745-9a29-2820f32e44d4.png" class="site-icon" width="32" height="32">

      <a href="https://dnschecker.org/#A/vatprism.org" target="_blank" rel="noopener nofollow ugc">DNS Checker</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/361;"><img src="https://user-images.githubusercontent.com/6569856/181076796-0479aabe-e03d-4e75-a170-dc01ab54e7e1.png" class="thumbnail" width="690" height="361" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/b/4/b4a37638514a6646e9d94b74f9e08a048d87bc5b_2_690x361.png, https://aws1.discourse-cdn.com/github/optimized/3X/b/4/b4a37638514a6646e9d94b74f9e08a048d87bc5b_2_1035x541.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/b/4/b4a37638514a6646e9d94b74f9e08a048d87bc5b.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/b/4/b4a37638514a6646e9d94b74f9e08a048d87bc5b_2_10x10.png"></div>

<h3><a href="https://dnschecker.org/#A/vatprism.org" target="_blank" rel="noopener nofollow ugc">DNS Checker - DNS Check Propagation Tool</a></h3>

  <p>Check DNS Propagation worldwide. DNS Checker provides name server propagation check instantly. Changed nameservers so do a DNS lookup and check if DNS and nameservers have propagated.</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://dnschecker.org/?#CNAME/www.vatprism.org">
  <header class="source">
      <img src="https://user-images.githubusercontent.com/6569856/181076797-85de0e61-09ce-4568-9188-0dea2bee79a5.png" class="site-icon" width="32" height="32">

      <a href="https://dnschecker.org/?#CNAME/www.vatprism.org" target="_blank" rel="noopener nofollow ugc">DNS Checker</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/361;"><img src="https://user-images.githubusercontent.com/6569856/181076798-ed8386f3-7674-4eba-8e4a-384696ffc689.png" class="thumbnail" width="690" height="361" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/b/4/b4a37638514a6646e9d94b74f9e08a048d87bc5b_2_690x361.png, https://aws1.discourse-cdn.com/github/optimized/3X/b/4/b4a37638514a6646e9d94b74f9e08a048d87bc5b_2_1035x541.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/b/4/b4a37638514a6646e9d94b74f9e08a048d87bc5b.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/b/4/b4a37638514a6646e9d94b74f9e08a048d87bc5b_2_10x10.png"></div>

<h3><a href="https://dnschecker.org/?#CNAME/www.vatprism.org" target="_blank" rel="noopener nofollow ugc">DNS Checker - DNS Check Propagation Tool</a></h3>

  <p>Check DNS Propagation worldwide. DNS Checker provides name server propagation check instantly. Changed nameservers so do a DNS lookup and check if DNS and nameservers have propagated.</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>What method are you using to query the DNS records for the domain?</p>

**Comment by yoannchaudet** [comment]
<p>We use the library we published here: <a href="https://github.com/github/pages-health-check" class="inline-onebox">GitHub - github/pages-health-check: Checks your GitHub Pages site for common DNS configuration issues</a>.</p>
<p>And we resolve the DNS records with both Cloudflare (1.1.1.1) and Google (8.8.8.8) if I am not mistaken.</p>
<p>In both cases I am unable to query your DNS zone (e.g. for A records):</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/0/2/028234182186b2bdcd3ebbfc39fe4378a337016b.png" data-download-href="/uploads/short-url/mbUHRYxCUeuhbOuVVPfftiIrrl.png?dl=1" title="image"><img src="https://user-images.githubusercontent.com/14911070/181076801-89ba7961-adbc-4a32-949a-d837fc6b07ac.png" alt="image" data-base62-sha1="mbUHRYxCUeuhbOuVVPfftiIrrl" width="458" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/0/2/028234182186b2bdcd3ebbfc39fe4378a337016b_2_458x500.png, https://aws1.discourse-cdn.com/github/optimized/3X/0/2/028234182186b2bdcd3ebbfc39fe4378a337016b_2_687x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/0/2/028234182186b2bdcd3ebbfc39fe4378a337016b_2_916x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/0/2/028234182186b2bdcd3ebbfc39fe4378a337016b_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1278×1394 239 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>I am not familiar with your registrar’s user interface but are you sure your DNS zone is properly published? Some registrar do have a concept of version you need to publish before propagation can start.</p>

**Comment by marvk** [comment, marked as answer]
<p>Unfortunately my provider wasn’t able to help me out, however, via</p>
<p>ttps://dns.google.com/query?<a href="http://name=vatprism.org" rel="noopener nofollow ugc">name=vatprism.org</a>&amp;rr_type=A&amp;ecs=</p>
<p><strong>I was able to determine that Google wasn’t accepting the DNS entry because of a DNSSEC issue:</strong></p>
<pre><code class="lang-auto">{
  "Status": 2,
  "TC": false,
  "RD": true,
  "RA": true,
  "AD": false,
  "CD": false,
  "Question": [
    {
      "name": "www.vatprism.org.",
      "type": 5
    }
  ],
  "Comment": "DNSSEC validation failure. Check http://dnsviz.net/d/www.vatprism.org/dnssec/ and http://dnssec-debugger.verisignlabs.com/www.vatprism.org for errors"
}
</code></pre>
<p><strong>After checking the “DNSSEC Status” checkbox on my providers page, the DNS issue went away and shortly after, GitHub started issuing a certificate.</strong></p>
<p>I’m not sure what caused Google to reject the DNS, since all of my other domains don’t have the checkbox checked and resolve without issue on Google.</p>
<p>While my case seems quite specific, I think the user experience on the GitHub Pages page could be improved:</p>
<p>First, it is not transparent when GitHub will try to reissue a certificate, forcing you to clear and re-add the domain as a workaround to manually trigger an attempt at reissuing.</p>
<p>Second, the warning being displayed is quite generic and I’m sure threads like this could be prevented by creating some official documentation.</p>
<p>It sounds like you guys are working on the whole certificate thing at the moment, so I’m sure many of these things are already being worked on.</p>
<p>Thanks for the otherwise excellent service!</p>

**Comment by yoannchaudet** [comment]
<p>Glad to hear you manage to get your certificate provisioned and thank you for the feedback!</p>
<p>We actually do try to obtain certificates on error continuously in the background. It is just not happening as fast as it does when you go to the Pages settings page simply to stay under rate limits imposed by our certificate provider (Let’s Encrypt).</p>
